lib/cocoapods-core/source/manager.rb

```diff
@@ -424,28 +426,48 @@ def canonic_url(url)
       def name_for_url(url)
         base_from_host_and_path = lambda do |host, path|
           if host && !host.empty?
-            base = host.split('.')[-2] || host
-            base += '-'
+            domain = PublicSuffix.parse(host) rescue nil
+            base = [domain&.sld || host]
+            base = [] if base == %w(github)
           else
-            base = ''
+            base = []
           end
 
-          base + path.gsub(/.git$/, '').gsub(%r{^/}, '').split('/').join('-')
+          path = path.gsub(/.git$/, '').gsub(%r{^/}, '').split('/')
+          path.delete('specs')
```

Comment:
should this be `path.pop if path.last == 'specs'`, to handle a `specs` segment in the middle of the path? can we add specs for that case?

Reply:
@segiddins gotcha - added!